contribs/dvrp/src/main/java/org/matsim/contrib/dvrp/path/VrpPaths.java

```diff
@@ -38,31 +39,55 @@
 	 */
 	public static VrpPathWithTravelData calcAndCreatePath(Link fromLink, Link toLink, double departureTime,
 			LeastCostPathCalculator router, TravelTime travelTime) {
-		Path path = fromLink == toLink ?
-				null :
-				router.calcLeastCostPath(fromLink.getToNode(), toLink.getFromNode(), departureTime + FIRST_LINK_TT,
+		Path path = fromLink == toLink ? null
+				: router.calcLeastCostPath(fromLink.getToNode(), toLink.getFromNode(), departureTime + FIRST_LINK_TT,
 						null, null);
 		return VrpPaths.createPath(fromLink, toLink, departureTime, path, travelTime);
 	}
 
-	public static VrpPathWithTravelData createZeroLengthPath(Link fromTolink, double departureTime) {
-		return new VrpPathWithTravelDataImpl(departureTime, 0, new Link[] { fromTolink }, new double[] { 0 });
+	public static VrpPathWithTravelData calcAndCreatePath(LinkTimePair diversionPoint, Link toLink,
```

Comment:
It is hard to differentiate which of method is meant for regular departures and which for diversion (the only difference the argument list: `LinkTimePair` vs `Link` and `double`)

Maybe rename to `calcAndCreatePathForDiversion()`?